sklearn/linear_model/tests/test_sgd.py

```diff
@@ -2144,3 +2165,21 @@ def test_loss_squared_loss_deprecated(Estimator):
         assert_allclose(est1.predict_proba(X), est2.predict_proba(X))
     else:
         assert_allclose(est1.predict(X), est2.predict(X))
+
+
+def test_quantile():
+    X, y = datasets.make_regression(
+        n_samples=1000, n_features=10, n_informative=9, random_state=1234
+    )
+    from .._quantile import QuantileRegressor
+
+    benchmark = QuantileRegressor(alpha=1e-6, solver="highs").fit(X, y).score(X, y)
```

Comment:
I assume that this will fail with older scipy because the solve is unkown

Reply:
Now that I am seeing the `alpha`, it makes me think that we only support `l1` penalty in the `QuantileRegressor`.

@lorentzenchr is it fine to support `l2` and `elasticnet` in SGD?

Reply:
Yes, this needs a `skip_if` for scipy >= 1.6, IIRC.

And yes, SGD already comes with L2 and L1+L2, so why not supporting it. It would be more code to forbid it:wink:

Reply:
Do we need test cases for other penalties?

Reply:
> And yes, SGD already comes with L2 and L1+L2, so why not supporting it. It would be more code to forbid it

@lorentzenchr so the reason to not support L2 and L1+L2 in `QuantileRegressor` is that we cannot reformulate the problem as a linear programming problem?

> Do we need test cases for other penalties?

Yes, it could be good. I don't know how easy this is but it could be great to have some integration test checking that the penalty has the expected impact on the `coef_` and in the same time that we are indeed fitting the expected quantile.

Reply:
I'm not sure on how to proceed further now, any pointers would be helpful. Thanks